blog-api-y-nguyen/Controllers/AuthorsController.cs

```diff
@@ -7,39 +7,43 @@
 using Microsoft.EntityFrameworkCore;
 using blog_api_y_nguyen.Models;
 using blog_api_y_nguyen.Repository;
+using blog_api_y_nguyen.Services;
+using AutoMapper;
 
 namespace blog_api_y_nguyen.Controllers
 {
     [Route("api/authors")]
     [ApiController]
     public class AuthorsController : ControllerBase
     {
-        private IAuthorRepository _authorRepository;
-        public AuthorsController(BlogContext context)
+        private readonly IAuthorService _authorService;
+        private readonly IMapper _autoMapper;
+        public AuthorsController(IMapper autoMapper, IAuthorService authorService)
         {
-            _authorRepository = new AuthorRepository(context);
+            _autoMapper = autoMapper;
+            _authorService = authorService;
         }
 
         // GET: api/Authors
         [HttpGet]
         public ActionResult<IEnumerable<Author>> GetAllAuthors()
         {
-            if ( _authorRepository.CheckAuthorsIsNull() )
+            if (_authorService.CheckAuthorsExist() == false)
             {
                 return NotFound();
             }
-            return _authorRepository.GetAllAuthors();
+            return Ok(_authorService.GetAllAuthors());
         }
 
         // GET: api/Authors/5
         [HttpGet("{id}")]
         public ActionResult<Author> GetAuthor(int id)
         {
-            if (_authorRepository.CheckAuthorsIsNull())
+            if (_authorService.CheckAuthorsExist() == false)
```

Comment:
no need to do this, the result of `_authorService.GetAuthor(id)` should already cover this check

Reply:
return Ok(author) :P